backend/Forum/Services/Implementations/TopicService.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Forum.Domain.Models;
+using Forum.Domain.ViewModels;
+using Forum.Repository.Interfaces;
+using Forum.Services.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Forum.Services.Implementations
+{
+    public class TopicService : ITopicService
+    {
+        private readonly ITopicRepository topicRepository;
+        private readonly ISubforumService subforumService;
+        private readonly IPostService postService;
+
+        public TopicService(ITopicRepository topicRepository, ISubforumService subforumService)
+        {
+            this.topicRepository = topicRepository;
+            this.subforumService = subforumService;
+        }
+
+        public TopicViewModel CreateTopic(Topic entity)
+        {
+            var topic = topicRepository.Insert(entity);
+            return FetchTopicViewModelWithPosts(topic);
+        }
+
+        public TopicViewModel DeleteTopic(Topic entity)
+        {
+            var topic = topicRepository.Delete(entity);
+            return FetchTopicViewModelWithPosts(topic);
+        }
+
+        public List<Topic> FetchAllTopics()
+        {
+            return topicRepository.FetchAllTopics().ToList();
+        }
+
+        public Topic FetchTopicById(Guid? id)
+        {
+            return topicRepository.FetchTopicById(id);
+        }
+
+        public List<TopicViewModel> FetchTopicsForSubforum(Guid subforumId)
+        {
+            var topics = subforumService.FetchSubforumById(subforumId).Topics;
+
+            var topicViewModels = new List<TopicViewModel>();
+
+            foreach (var t in topics)
+            {
+                topicViewModels.Add(FetchTopicViewModelWithPosts(t));
+            }
+
+            return topicViewModels;
+        }
+
+        public TopicViewModel FetchTopicViewModelWithPosts(Topic entity)
+        {
+            var posts = entity.Posts;
+            var postViewModels = new List<PostViewModel>();
+
+            foreach (var p in posts)
+            {
+                postViewModels.Add(postService.FetchPostViewModel(p.Id));
+            }
```

Comment:
This can be converted to LINQ synthax:
```suggestion
            var postViewModels = posts.Select(p => postService.FetchPostViewModel(p.Id)).ToList();
```